docker/Dockerfile

```diff
@@ -1,34 +1,33 @@
-FROM alpine
-MAINTAINER Marios Andreopoulos <marios@landoop.com>
+# Build From Source
+FROM docker.yektanet.tech/node:11-alpine
+RUN apk update && apk add git
+RUN git clone https://github.com/yn360/schema-registry-ui
+WORKDIR /schema-registry-ui
+RUN npm install && npm run build-prod
 
+# Build Image
+FROM docker.yektanet.tech/alpine
 WORKDIR /
+
 # Add needed tools
-RUN apk add --no-cache ca-certificates wget \
-    && echo "progress = dot:giga" | tee /etc/wgetrc
+RUN apk add --no-cache ca-certificates wget && \
+    echo "progress = dot:giga" | tee /etc/wgetrc
 
 # Add and Setup Caddy webserver
-RUN wget "https://github.com/mholt/caddy/releases/download/v0.10.11/caddy_v0.10.11_linux_amd64.tar.gz" -O /caddy.tgz \
-    && mkdir caddy \
-    && tar xzf caddy.tgz -C /caddy --no-same-owner \
-    && rm -f /caddy.tgz
-
-# Add and Setup Schema-Registry-Ui
-ENV SCHEMA_REGISTRY_UI_VERSION="0.9.5"
-RUN wget "https://github.com/Landoop/schema-registry-ui/releases/download/v.${SCHEMA_REGISTRY_UI_VERSION}/schema-registry-ui-${SCHEMA_REGISTRY_UI_VERSION}.tar.gz" \
-         -O /schema-registry-ui.tar.gz \
-    && mkdir /schema-registry-ui \
-    && tar xzf /schema-registry-ui.tar.gz -C /schema-registry-ui --no-same-owner \
-    && rm -f /schema-registry-ui.tar.gz \
-    && rm -f /schema-registry-ui/env.js \
-    && ln -s /tmp/env.js /schema-registry-ui/env.js
+ENV CADDY_VERSION="0.10.11"
+RUN wget "https://github.com/mholt/caddy/releases/download/v${CADDY_VERSION}/caddy_v${CADDY_VERSION}_linux_amd64.tar.gz" -O /caddy.tgz && \
+    mkdir caddy && \
+    tar xzf caddy.tgz -C /caddy --no-same-owner && \
+    rm -f /caddy.tgz
 
-# Add configuration and runtime files
-ADD Caddyfile /caddy/Caddyfile.template
-ADD run.sh /
+# Add Schema-Registry-UI, configurations, and runtime files
+COPY --from=0 /schema-registry-ui/dist/ /schema-registry-ui
+RUN rm -f /schema-registry-ui/env.js && ln -s /tmp/env.js /schema-registry-ui/env.js
+COPY --from=0 /schema-registry-ui/docker/Caddyfile /caddy/Caddyfile.template
```

Comment:
I guess we can copy this from the current context as well instead of the previous stage.